### PR TITLE
Refactor HitL software for composability

### PR DESCRIPTION
**Pull request by jack-greenberg**

### CODE REVIEW

**Comment by jack-greenberg** [comment]
Removed old HitL files because they were causing issues. Linter is complaining, but we can safely ignore it because it's trying to lint files that were deleted.

File: projects/hitl/lib/hitl.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+        for pin in pins:
+            self.register_pin(pin["name"], pin["pintype"], pin["number"], pin["dir"])
+
+        self.log.info("HitL initialization complete.")
+
+    def close(self):
+        self.i2c.close()
+
+        if self.can:
```
Comment (jack-greenberg) [inline projects/hitl/lib/hitl.py:95]: Need to use getattr

File: projects/hitl/lib/tests/test_adc.py
Comment (jack-greenberg) [inline projects/hitl/lib/tests/test_adc.py:1]: Add tests or remove

File: projects/hitl/lib/tests/test_dac.py
Comment (jack-greenberg) [inline projects/hitl/lib/tests/test_dac.py:1]: More tests?

File: third_party/requirements.txt
Comment (jack-greenberg) [inline third_party/requirements.txt:1]: Add versions

File: vehicle/mkv/software/air_control/tests/BUILD [hunk 2]
```diff
@@ -11,11 +11,13 @@ pytest_test(
...
         "//vehicle/mkv/software/air_control",
     ],
+    deprecation = "Old HitL library not around anymore; do not use this",
+    tags = ["manual"],
     target_compatible_with = [
         "//bazel/constraints:atmega16m1",
     ],
     deps = [
-        "//projects/hitl/software/hitl",
+        # "//projects/hitl/software/hitl",
```
Comment (jack-greenberg) [inline vehicle/mkv/software/air_control/tests/BUILD:20]: rm

File: vehicle/mkvi/hardware/bms/bms_core/bms_core.kicad_pro
Comment (jack-greenberg) [inline vehicle/mkvi/hardware/bms/bms_core/bms_core.kicad_pro:1]: Revert

File: vehicle/mkvi/software/tests/BUILD
Comment (jack-greenberg) [inline vehicle/mkvi/software/tests/BUILD:1]: Edit

File: vehicle/mkvi/software/tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+def canbus():
+    bus = Bus(
+        channel = "can0",
+        bustype = "socketcan",
+        bitrate = 500000
+    )
+
+    yield bus
+
+    # bus.shutdown()
```
Comment (jack-greenberg) [inline vehicle/mkvi/software/tests/conftest.py:26]: todo

File: vehicle/mkvi/software/tests/conftest.py [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+        "tunable_id": 0x6ea,
+        "binary": "vehicle/mkvi/software/throttle/throttle_patched.bin",
+    },
+    {
+        "name": "brakes",
+        "btldr_id": 0x718,
+        "tunable_id": 0x6e6,
+        "binary": "vehicle/mkvi/software/brakes/brakes_patched.bin",
+    },
+]
```
Comment (jack-greenberg) [inline vehicle/mkvi/software/tests/conftest.py:42]: Add

File: vehicle/mkvi/software/tests/conftest.py
Comment (jack-greenberg) [inline vehicle/mkvi/software/tests/conftest.py:1]: Clarify that these tests are to be used with RKH

File: vehicle/mkvi/software/tests/conftest.py [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+
+    # Flash
+    # for ecu in ECUS:
+    #     resp = btldr.ping(ecu["btldr_id"], 1)
+    #
+    #     if resp:
+    #         log.info("Found {}, flashing.".format(ecu["name"]))
+    #         btldr.flash(ecu["btldr_id"], ecu["binary"], 1)
+    #     else:
+    #         log.warning("Unable to ping {}. Is it connected?".format(ecu["name"]))
```
Comment (jack-greenberg) [inline vehicle/mkvi/software/tests/conftest.py:64]: Todo?

File: vehicle/mkvi/software/tests/test_rtd.py
Comment (jack-greenberg) [inline vehicle/mkvi/software/tests/test_rtd.py:1]: Edit

**Review by amit-kumarh — APPROVED** [review]
Black error due to deleted python file bug, compilation error bc dashboard is not merged in.

**Pull request merged**